1.sh

```diff
@@ -1,35 +1,34 @@
 #! /bin/sh
 
-if [-d $1] && [-d $2]
+if [ -d "$1" ] && [ -d "$2" ]
 then
 	echo "Dirs exist"
-elif [! -d $2] 
+elif [ ! -d "$2" ] 
 	echo "Flash path doesn't exist"
-	mkdir $2
+	mkdir "$2"
 fi
 
-for file in $1/*
+for file in "$1"/*
 do
-	echo $(basename $file)
-	flashfile=$($2/$(basename $file))
-	if [! -f $flashfile]; then
-		echo "Copying files to $2"
-		cp $file $2
+	echo "$(basename $file)"
+	flashfile="$2/$(basename $file)"
+
+	if [ ! -f "$flashfile" ]; then
+		echo "Copying files to '$2'"
+		cp "$file" "$2"
```

Comment:
Again, protect your variables, and even your strings! In the case, the outcome could be quite dangerous. (data loss due to overwrites)

Here's an example situation:

I want to copy file `My First File` which is in the current directory, to the directory `/home/user/`. Your way would be as follows:

```bash
cp My First File /home/user/
```

This would, however, move files `My`, `First`, and `File` to `/home/user/`, overwriting anything with those names in the target directory. The correct way, therefore, would be, at the very least:

```bash
cp 'My First File' /home/user/
```

Since `/home/user/` has no spaces, variables, or any special characters bash would otherwise interpret, you're safe to leave it as-is. The `My First File`, however has spaces, as so needs to be protected by either single or double quotes; in this case, it doesn't matter which.